tools/mafft.py

```diff
@@ -0,0 +1,157 @@
+            mafft_archive_extension = get_mafft_archive_extension(mafft_os)
+            binaryPath              = get_mafft_binary_path(mafft_os, mafft_bitdepth)
+            binaryDir               = get_mafft_binary_path(mafft_os, mafft_bitdepth, full=False)
+            
+            target_rel_path = '{binPath}'.format(ver=tool_version, os=mafft_os, binPath=binaryPath)
+            verify_command  = 'cd {dir}/mafft-{ver}/{binDir} && {dir}/mafft-{ver}/{binPath} --version > /dev/null 2>&1'.format(dir=util.file.get_build_path(), ver=tool_version, os=mafft_os, binPath=binaryPath, binDir=binaryDir) 
+            destination_dir = '{dir}/mafft-{ver}'.format(dir=util.file.get_build_path(), ver=tool_version, os=mafft_os, binPath=binaryPath)
+
+            install_methods.append(
+                    tools.DownloadPackage( url.format(ver=tool_version, os=mafft_os, ext=mafft_archive_extension),
+                    target_rel_path=target_rel_path,
+                    destination_dir=destination_dir,
+                    verifycmd=verify_command))
+
+        tools.Tool.__init__(self, install_methods = install_methods)
+
+    def version(self):
+        return tool_version
+
+    def execute(self, inFastas, outFile, localpair, globalpair, preservecase, reorder, 
+                outputAsClustal, maxiters, gapOpeningPenalty=None, offset=None, threads=1, verbose=True):
+
+        inputFileName         = ""
+        tempCombinedInputFile = ""
+
+        # get the full paths of input and output files in case the user has specified relative paths
+        inputFiles = []
+        for f in inFastas:
+            inputFiles.append(os.path.abspath(f))
+        outFile = os.path.abspath(outFile)
+
+        # if multiple fasta files are specified for input
+        if len(inputFiles)>1:
+            # combined specified input files into a single temp FASTA file so MAFFT can read them
+            tempFileSuffix = ""
+            for filePath in inputFiles:
+                tempFileSuffix += "__" + os.path.basename(filePath)
+            tempCombinedInputFile = util.file.mkstempfname('__combined.{}'.format(tempFileSuffix))
+            with open(tempCombinedInputFile, "wb") as outfile:
+                for f in inputFiles:
+                    with open(f, "rb") as infile:
+                        outfile.write(infile.read())
+                #outFile.close()
+            inputFileName = tempCombinedInputFile
+        # if there is only once file specified, just use it
+        else:
+            inputFileName = inputFiles[0]
+
+        # change the pwd, since the shell script that comes with mafft depends on the pwd
+        # being correct
+        pwdBeforeMafft = os.getcwd()
+        os.chdir(os.path.dirname(self.install_and_get_path()))
+
+        # build the MAFFT command
+        toolCmd = [self.install_and_get_path()]
+
+        toolCmd.append("--auto")
+        toolCmd.append("--thread {}".format( max( int(threads), 1 )) )
```

Comment:
according to MAFFT web site, a value of -1 is how to ask MAFFT to auto-detect the available cores, so we should probably allow a value of -1 (perhaps even default to it), but not allow a value of zero or other negative values.